### DISCUSSION TITLE
"Cannot build name" in relationships

### DISCUSSION THREAD

**Discussion by lukuser** [Budibase/budibase#8901] [Help]
I have an Azure DataSource with several tables and mulstiple relationships. In the process of defining the relationships, I somehow managed to accumulate some entries that say "Cannot build name" on the table side with one of the columns listed as "undefined". I cannot delete those rows. Attempt to complete and then delete don't work ("Delete is enabled, and I get green "Successfull" status, but the relationship is still listed). Saving it after providing the missing config just creates a new relationship leaving this one still listed.

![image](https://user-images.githubusercontent.com/81827963/205172488-c9756f98-6163-4ed9-bab6-29acf104fe95.png)

**Comment by melohagan** [comment]
Hey @lukuser 

Are you able to remove the datasource and re-add it, then fetch the tables?

E.g. 
![Screenshot 2022-12-07 at 14 04 05](https://user-images.githubusercontent.com/101575380/206199515-9f8ae188-bbc9-4a76-9eb0-14b11dcc11d1.png)
